src/Api/ChangeStream/ResourceStack.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the puli/repository package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Puli\Repository\Api\ChangeStream;
+
+use Puli\Repository\Api\Resource\PuliResource;
+
+/**
+ * Represents a versionned stack of resources for a given path.
+ *
+ * You can access different versions of a resource using a ChangeStream
+ * that will return you resource stacks.
+ *
+ * @since  1.0
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+interface ResourceStack
```

Comment:
Given that the `ResourceStack` is a basic domain object that is not likely to be replaced, I think it would be sufficient to simply add the implementation here (just like with `ResourceMetadata`). This also enables us to add more methods, should we need any.